### ISSUE TITLE
brainstorming anti-labels as label deletion hints:

### ISSUE THREAD

**Issue by dobkeratops** [bbernhard/imagemonkey-core#182]
sometimes i see cases where the autocomplete removed a comma (or other errors),
I figured dropping a label ```~foo``` could become a hint that the label ```foo``` is erroneous

<img width="724" alt="screen shot 2018-08-01 at 13 59 15" src="https://user-images.githubusercontent.com/1120754/43523065-19bfc530-9593-11e8-8e6b-c2ee83a7785a.png">

usually these cases produce nonsensical words, so they should be quite clear already.. still I figure something explicit would clarify

**Comment by bbernhard** [comment]
You are right, even as a moderator it's not possible to remove non-productive labels. While implementing the remove functionality for moderators, I thought that such a functionality will only be needed for productive labels. But you are right, it makes sense to also add that for non-productive labels. 

I guess typos are only an issue for non-productive labels, right? What do you think about this (more generalized) approach then:

In order to add a non-productive label you need to be signed up. That means we already know which user added which label to which image. So we could allow users to remove the labels they've added. i.e: If you've added a non-productive label to an image, you can delete it again. I would go even a step further and allow a "real" delete here, instead of the "fake delete" that happens when a moderator "removes" a productive label(in case of a "fake delete" we are just voting with a much greater voting power for "invalid", which results in a struckthrough label).

Furthermore moderators should be able to remove every non-productive label from an image. I would really restrict that to non-productive labels here, as I would see the delete function only as a way to remove misspelled/spam labels. If the label in general is fine, but just wrong for the given image, I would go with the voting approach to invalidate the label. That way the label won't get deleted, but just gets struckthrough (I kind of like that, as it serves as some sort of "label history")


 > I figured dropping a label ~foo could become a hint that the label foo is erroneous

I like the idea to mark the labels somehow, until we have a proper solution in place. The only thing I am not sure about is, if the parser behaves nicely when the `~` operator is also part of the label. (this reminds me, that it's probably ain't a bad idea to forbid `&`, `|` and `~` in the label name. Even if it's not a problem now, it could make some problems when the parsing logic evolves). 

In case you are marking the erroneous labels somehow, please let me know which syntax you are using. Once we have a proper solution in place, I can manually delete all the erroneous labels from the database. :)

**Comment by dobkeratops** [comment]
>>  So we could allow users to remove the labels they've added. 

that makes sense. maybe eventually there can be browse filters for your own contributions to help double check etc

>> The only thing I am not sure about is, if the parser behaves nicely when the ~ operator is also part of the label. 

right; i'm also uneasy about the reuse of ~ vs label arithmetic, and other suggestions of a "confirmed absence" label. I've only used this a few times. I was hoping it could be corrected manually as a last resort.. these markers could be removed aswell.
